### PR TITLE
Ported `_log_softmax` to the new API

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Ported `_log_softmax` to the new API

### CODE REVIEW

**Comment by Chillee** [comment]
Out of curiosity, does `REDUCTION_BOXED` work for this?

**Comment by vfdev-5** [comment]
@Chillee seems like with `REDUCTION_BOXED` also working. Let me update to use it.

File: functorch/csrc/BatchRulesReduceOps.cpp [hunk 1]
```diff
@@ -201,6 +201,17 @@ static std::tuple<Tensor, Tensor, Tensor> expand_bdims(
       c_has_bdim ? c : c.expand_as(flagpole));
 }
 
+std::tuple<Tensor,optional<int64_t>> _log_softmax_batch_rule(
```
Comment (Chillee) [inline functorch/csrc/BatchRulesReduceOps.cpp:204]: Delete the old code and it's good to go :)

**Review by Chillee — APPROVED** [review]

**Comment by Chillee** [comment]
Will merge after CI passes.

**Pull request merged**